scan/lib/scan/test_command_generator.rb

```diff
@@ -49,7 +49,9 @@ def actions
         actions = []
         actions << :clean if config[:clean]
         actions << :build unless config[:skip_build]
-        actions << :test
+        actions << "build-for-testing" if config[:build_for_testing]
+        actions << :test unless config[:test_without_building] || config[:build_for_testing]
+        actions << "test-without-building" if config[:test_without_building]
```

Comment:
you are right - i splitted it, and well - i learned my lesson - will use unless not to often anymore 🙏